pkg/config/con.go

```
@@ -0,0 +1,947 @@
+		dir := filepath.Dir(file)
+		baseName := strings.TrimSuffix(filepath.Base(file), filepath.Ext(file))
+		ext := filepath.Ext(file)
+
+		// Construct a unique key for the folder + base name
+		key := filepath.Join(dir, baseName)
+
+		// Assign .yaml as priority if it exists, or fallback to .yml
+		if existingFile, exists := priorityMap[key]; exists {
+			if ext == ".yaml" {
+				priorityMap[key] = file // Replace .yml with .yaml
+			} else if ext == ".yml" && strings.HasSuffix(existingFile, ".yaml") {
```

golangci-lint (1.62.2)

SA4017: HasSuffix doesn't have side effects and its return value is ignored

(staticcheck)